### PR TITLE
Cancel node context after StopAll

### PR DESCRIPTION
**Pull request by rkapka**
**What type of PR is this?**

Other

**What does this PR do? Why is it needed?**

#7625 removed context cancelling from the beacon node because it caused a panic - cancelling the parent context before `StopAll`, during which services are being shut down, caused child contexts inside these services to be cancelled before services had a chance to clean themselves up.

Last week me and @prestonvanloon discussed some related issue and it occurred to us that it's safe to cancel the parent context *after* all child contexts have already done their job, i.e. after `StopAll`. This PR adopts this solution. 

**Which issues(s) does this PR fix?**

N/A

**Other notes for review**
So simple yet so clever.

### CODE REVIEW

File: slasher/node/node.go [hunk 1]
```diff
@@ -150,11 +150,11 @@ func (s *SlasherNode) Close() {
 	defer s.lock.Unlock()
 
 	log.Info("Stopping hash slinging slasher")
-	s.cancel()
```
Comment (rkapka) [inline slasher/node/node.go:153]: The same bug-prone code as in the beacon node - cancelling the parent context before `StopAll` is called.

**Review by rauljordan — APPROVED** [review]

**Comment by rauljordan** [comment]
Conflict and tests failing

**Review by prestonvanloon — DISMISSED** [review]

**Comment by rkapka** [comment]
> Conflict and tests failing

It's the duplicate wallet import test that is failing. It's causing problems in all of our PRs.

**Review by terencechain — COMMENTED** [review]
How about validator?

https://github.com/prysmaticlabs/prysm/blob/master/validator/node/node.go#L151

**Comment by rkapka** [comment]
> How about validator?
> 
> https://github.com/prysmaticlabs/prysm/blob/master/validator/node/node.go#L151

I had not implemented this for validator because the struct did not contain `ctx` and `cancel`. I agree it's better to have consistency across all modules, so I added the relevant parts.

**Review by rauljordan — APPROVED** [review]

**Pull request merged**